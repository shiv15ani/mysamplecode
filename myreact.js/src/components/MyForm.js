import React ,  {Component,Fragment }from 'react';
import InputText from './InputText';
import Dropdown from './Dropdown';
import DisplayComponent from './DisplayComponent';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {add, submitAll, display, loadData} from '../actions/AddFields';
import {Link} from 'react-router-dom';
import MyContainer from '../containers/MyContainer'
class MyForm extends Component{
	


	Submit(e){
		this.props.submitAll(this.props.data);
	}
	render(){
		
		var requirement_type_val=['New/Immediate','Future','Replacement'];
		var owner=['RMG','TAG'];
		var exp_range=['0-2','2-4','4-6','6-8','8-10','10-12','12+'];
		var grp=['T&D','MSO','Digital']
		return(
				<Fragment>
					<MyContainer/>
					<div  className="container">
						<div>
								<h2>Add Requirement</h2>
						</div>
						<form >
								
							 <div className="form-row">
								<InputText id='crm_id' data={this.props.data} add={this.props.addTask}>CRM ID</InputText>
								<InputText id='crm_name' data={this.props.data} add={this.props.addTask}>CRM Name</InputText>
							 </div>	
							  <div className="form-row">
								<InputText id='request_id' data={this.props.data} add={this.props.addTask}>Request ID</InputText>
								<InputText id='requirement_id' data={this.props.data} add={this.props.addTask}>Requirement ID</InputText>
							 </div>
							 <div className="form-row">	
								<Dropdown id='requirement_type' data={this.props.data} add={this.props.addTask} values={requirement_type_val}>Requirement Type</Dropdown>
							</div>
							<div className="mycenter"><Link to="/DisplayComponent" ><input type="submit" className="btn btn-success" value="SUBMIT"  onClick={this.Submit.bind(this)}/></Link></div>
							 
						</form>
					</div>
					
				</Fragment>
			)
	}
}
const  mapStateToProps = (state) => {
	console.log("form")
  console.log(state);
  return{
    data: state.Reducer
  }
}
const mapDispatchToProps = (dispatch) => {
  return{
    addTask: bindActionCreators(add, dispatch),
    submitAll: bindActionCreators(submitAll, dispatch),
   	loadData: bindActionCreators(loadData, dispatch)

  }
}
export default connect(mapStateToProps, mapDispatchToProps)(MyForm);