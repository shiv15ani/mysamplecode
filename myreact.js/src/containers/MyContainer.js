import React ,{Component} from 'react';
import LoginPage from '../components/LoginPage';
import MyForm from '../components/MyForm';
import '../css/Bookstore.css'
import {BrowserRouter,Link} from 'react-router-dom';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';


export default class MyContainer extends Component{
	
	render(){
		return(
			<div>
			
		<nav className="navbar navbar-light bg-light navbar-expand-sm">
			<Link to='/' className="navbar-brand">Resource Management</Link>
			<button className="navbar-toggler" data-toggle="collapse" data-target="#navbarMenu">
			<span className="navbar-toggler-icon"></span>
			</button>
			<div className="collapse navbar-collapse" id="navbarMenu">
					<ul className="navbar-nav nav-tabs nav-fill">
					<li className="nav-item"> 
							<Link to='/LoginPage'>
								<div className="nav-link" >Login</div>
							</Link>
					</li>
					
					<li className="nav-item">
							<Link to='/MyForm'>
								<div className="nav-link " >ADD REQUIREMENT</div>
							</Link>
				    </li>
				    <li className="nav-item">
				    	<Link to='/DisplayComponent'>
							<div className="nav-link" >VIEW REQUIREMENT</div>
						</Link>
				    </li>
				    <li className="nav-item">
				    		<Link to='/ProfileForm'>
								<div className="nav-link" >ADD PROFILE</div>
							</Link>
				    </li>
				    <li className="nav-item">
				    		<Link to='/DisplayProfile'>
								<div className="nav-link" >VIEW PROFILE</div>
							</Link>
				    </li>
					</ul>
			</div>
		
		
		
		
		
		
			</nav>
			</div>
			)
	}
}
