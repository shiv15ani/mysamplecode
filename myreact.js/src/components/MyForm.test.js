import React from 'react';
import ReactDOM from 'react-dom';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import {shallow} from 'enzyme';
import MyForm from './MyForm';
import MyContainer from '../containers/MyContainer';
import toJson from 'enzyme-to-json';
const mockStore = configureStore();

configure({ adapter: new Adapter() });
test('testing my component',()=>{
	const wrapper=shallow(<MyForm store={mockStore()} />).dive();
	console.log(wrapper)
	expect(wrapper.contains("Add Requirement")).toBe(true);
	
})
test('<MyForm>',()=>{
	const wrapper=shallow(<MyForm store={mockStore({ runtime: {} })} />).dive();
	expect(wrapper.length).toEqual(1);
	
})
test('testing MyContainer component exists or not',()=>{
	const wrapper=shallow(<MyForm store={mockStore({ runtime: {} })} />).dive();
	expect(wrapper.find(MyContainer).exists());
	
})
test('testing MyContainer component length',()=>{
	const wrapper=shallow(<MyForm store={mockStore({ runtime: {} })} />).dive();
	expect(wrapper.find(MyContainer).length).toEqual(1);
	
})
it('testing css selector',()=>{
	const wrapper=shallow(<MyForm store={mockStore({ runtime: {} })} />).dive();
	expect(wrapper.find('.form-row').length).toBeGreaterThan(2);
	
})

test('to view the tree in console',()=>{
	const wrapper=shallow(<MyForm store={mockStore({ runtime: {} })} />).dive();
	const tree=toJson(wrapper);
	//console.log(tree);
})
test('to view the snapshot ',()=>{
	const wrapper=shallow(<MyForm store={mockStore({ runtime: {} })} />).dive();
	const tree=toJson(wrapper);
	expect(tree).toMatchSnapshot();
})
