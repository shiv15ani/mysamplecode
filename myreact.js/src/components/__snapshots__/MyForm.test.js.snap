// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to view the snapshot  1`] = `
<React.Fragment>
  <Bookstore />
  <div
    className="container"
  >
    <div>
      <h2>
        Add Requirement
      </h2>
    </div>
    <form>
      <div
        className="form-row"
      >
        <InputText
          add={[Function]}
          id="crm_id"
        >
          CRM iD
        </InputText>
        <InputText
          add={[Function]}
          id="crm_name"
        >
          CRM Name
        </InputText>
      </div>
      <div
        className="form-row"
      >
        <InputText
          add={[Function]}
          id="request_id"
        >
          Request ID
        </InputText>
        <InputText
          add={[Function]}
          id="requirement_id"
        >
          Requirement ID
        </InputText>
      </div>
      <div
        className="form-row"
      >
        <Dropdown
          add={[Function]}
          id="requirement_type"
          values={
            Array [
              "New/Immediate",
              "Future",
              "Replacement",
            ]
          }
        >
          Requirement Type
        </Dropdown>
      </div>
      <div
        className="mycenter"
      >
        <Link
          replace={false}
          to="/DisplayComponent"
        >
          <input
            className="btn btn-success"
            onClick={[Function]}
            type="submit"
            value="SUBMIT"
          />
        </Link>
      </div>
    </form>
  </div>
</React.Fragment>
`;
